src/models/User.ts

```diff
@@ -175,7 +175,7 @@ export class User {
       }
     } else {
       formatLogMessage(logger)?.debug(`Invalid user: ${this.email}`);
-      return null;
+      return this;
```

Comment:
We need to return the user object, because the signupController is expecting the user object with its errors, so that this info can be passed to the frontend.